Library/test-helpers/lib.sh

```diff
@@ -843,6 +843,88 @@ limeStopTPMEmulator() {
     esac
 }
 
+true <<'=cut'
+=pod
+
+=head2 limeCheckRemotePort
+
+Use limeCheckRemotePort to check port on specified ip adress.
+
+    limeCheckRemotePort [LOGFILE] [PORT_NUMBER] [IP]
+
+=over
+
+=item
+
+    PORT - Port number to check it.
+
+=item
+
+    IP - IP adress to specify host.
+
+=back
+
+Returns 0 when the start was successful, non-zero otherwise.
+
+=cut
+
+function limeCheckRemotePort() {
+
+    PORT=$1
+    IP=$2
+
+    echo $IP | grep -q "::" && NMAP_PARAMS="-6"
+
+    nmap -p $PORT $IP $NMAP_PARAMS | grep -E -q "^${PORT}/(tcp|udp) open"
+}
+
+true <<'=cut'
+=pod
+
+=head2 limeWaitForKeylimeService
+
+Use rlWaitForSocket to wait for the services to start or if it's remote,
+use limeCheckRemotePort for check open ports.
+
+    limeWaitForKeylimeService [LOGFILE] [PORT_NUMBER] [IP]
+
+=over
+
+=item
+
+    LOGFILE - Filepath to a log file.
+
+=item
+
+    PORT - Port number to wait for.
+
+=item
+
+    IP - IP adress to wait for.
+
+=back
+
+Returns 0 when the start was successful, non-zero otherwise.
+
+=cut
+
+limeWaitForKeylimeService() {
+
+    local LOGFILE=$1
+    local PORT=$2
+    local IP=$3
+
+    if [ -z "${IP}" ]; then
+        if ! rlWaitForSocket $PORT -d 0.5 -t ${limeTIMEOUT}; then
+            cat $LOGFILE
+            return 1
+        else
+            return 0
+        fi
+    elif [ -n "${IP}" ]; then
+        rlWaitForCmd "limeCheckRemotePort ${PORT} ${IP}" -m 3 -t 3 -d 1
```

Comment:
use `-t ${limeTIMEOUT}` instead of `-m 3 -t 3 -d 1` so it is in par with the command above. Sane default for `d` is probably 1s (since it is network connection).

Reply:
Also, maybe close port and IP in quotes to avoid unwanted expansion by shell, i.e.
```
rlWaitForCmd "limeCheckRemotePort '${PORT}' '${IP}'" -d 1 -t ${limeTIMEOUT}
```